Geometry/MuonNumbering/src/GEMNumberingScheme.cc

```diff
@@ -59,31 +59,40 @@ int GEMNumberingScheme::baseNumberToUnitNumber(const MuonBaseNumber& num) {
   //  ring    = num.getSuperNo(theRingLevel);
   // GEM are only on the first ring
   ring = 1;
-  station = num.getSuperNo(theStationLevel);
+
+  // GE0 has the layer encoded in the ring level
+  if (num.getBaseNo(theRingLevel) == 0) {  // 0 => GE1/1, GE2/1
+    station = num.getSuperNo(theStationLevel);
 #ifdef LOCAL_DEBUG
-  edm::LogVerbatim("GEMNumberingScheme") << "GEMNumbering: Ring " << ring << " Station "
-                                         << num.getSuperNo(theStationLevel) << ":" << station;
+    edm::LogVerbatim("GEMNumberingScheme")
+        << "GEMNumbering: Ring " << ring << " Station " << num.getSuperNo(theStationLevel) << ":" << station;
 #endif
 
-  roll = num.getBaseNo(theRollLevel) + 1;
-  const int copyno = num.getBaseNo(theSectorLevel) + 1;
-  if (copyno < 50) {
-    if (copyno % 2 == 0) {
-      layer = 2;
-      chamber = copyno - 1;
-    } else {
-      layer = 1;
-      chamber = copyno;
-    }
-  } else {
-    int copynp = copyno - 50;
-    if (copynp % 2 != 0) {
-      layer = 2;
-      chamber = copynp - 1;
+    roll = num.getBaseNo(theRollLevel) + 1;
+    const int copyno = num.getBaseNo(theSectorLevel) + 1;
+    if (copyno < 50) {
```

Comment:
magic number should be named constant

Reply:
This was some code I rearranged, so I'm not sure where to derive it from. @bsunanda can this be taken from the xml or so? I can't find anywhere with a '50' to show the demarcation of layer 1/2 chambers, it seems implicit. In which case shall I just introduce a `layerDemarcation` (is there a better name?) variable in the code itself?

Reply:
I think layerDemarcation may be a good name. May be I shall introduce this in the xml file as well for future usage.